### PR TITLE
Drop six from third_party

### PR DESCRIPTION
**Pull request by aiuto**
Drop six from third_party.  We don't use it to build or test bazel.  This potentially could break a user who depended on it from bazel_tools.

#18272  - part 1

### CODE REVIEW

File: src/BUILD [hunk 1]
```diff
@@ -145,7 +144,7 @@ rule_size_test(
     # WARNING: Only adjust the number in `expect` if you are intentionally
     # adding or removing embedded tools. Know that the more embedded tools there
     # are in Bazel, the bigger the binary becomes and the slower Bazel starts.
-    expect = 431,
+    expect = 445,
```
Comment (aiuto) [inline src/BUILD:147]: The diff is actually 465 to 445.  The 431 you may be seeing on the left is from an intermediate step where I brought it down too much.

Comment (Wyverald) [reply src/BUILD:147]: 431 is actually the number at HEAD. you should probably adjust this number further

Comment (aiuto) [reply src/BUILD]: Yes. I see that now Alex's removal of the android tools happened after my last sync from head, so I was based off the old value.

**Review by meteorcloudy — APPROVED** [review]

**Comment by meteorcloudy** [comment]
We need to import the non-third_party change as a CL first. @aiuto Can you split that into a separate PR?

**Comment by aiuto** [comment]
Done #23896

**Comment by aiuto** [comment]
This is now pure third_party.
I am not sure why the lock file is showing up. It is not changed.

**Comment by iancha1992** [comment]
@aiuto  Could you plz squash all the commits into one? Also, the presumits are not passing. Is that okay?

**Comment by aiuto** [comment]
Done. It's all rebased to one commit.

**Pull request closed without merging**